src/main/java/utilities/Utils.java

```diff
@@ -179,8 +153,8 @@ public static void swap(int[] input, int a, int b) {
     public static double normalise(double a_value, double a_min, double a_max) {
         if (a_min < a_max)
             return (a_value - a_min) / (a_max - a_min);
-        else    // if bounds are invalid, then return same value
-            return a_value;
+        else    // if bounds are invalid, then return 0.0
+            return 0.0;
```

Comment:
I'm not sure if this makes sense. If you're trying to normalise 5 between [7,10], should you get the same result as if you were trying to normalise 20 in the same interval? Probably best actually to raise an error...

Reply:
Agreed. Done.